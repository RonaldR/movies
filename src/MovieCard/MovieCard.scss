.movie-card {
	width: 100%;
}

.movie-card__link {
	display: flex;
	padding: 16px;
	background: #fafafa;
	border-bottom: 1px solid #e8e8e8;
	text-decoration: none;
	color: green;
	transition: 0.2s;
	box-shadow: none;

	&:hover {
		transform: scale(1.05);
		box-shadow: 1px 2px 7px rgba(0, 0, 0, 0.2);

		.movie-card__poster {
			transform: scale(1.1);
		}
	}
}

.movie-card__poster {
	width: 120px;
	margin-right: 12px;
	transition: 0.3s;

	img {
		width: 100%;
		border-radius: 3px;
	}
}
