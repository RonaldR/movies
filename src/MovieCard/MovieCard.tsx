import React from 'react';
import { Link } from 'react-router-dom';
import { MovieSearch } from '../types/types';

import './MovieCard.scss';

type Props = {
    movie: MovieSearch
}

const MovieCard = ({ movie }: Props) => (
    <li key={movie.imdbID} className="movie-card">
        <Link to={`/detail/${movie.imdbID}`} className="movie-card__link">
            <figure className="movie-card__poster">
                <img src={movie.Poster} alt="" />
            </figure>
            <div className="movie-card__title">
                {movie.Title} ({movie.Year})
            </div>
        </Link>
    </li >
);

export default MovieCard;
