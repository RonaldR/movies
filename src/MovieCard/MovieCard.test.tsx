import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MovieCard from './MovieCard';

test('movie card', async () => {
    const { asFragment } = render(<MemoryRouter><MovieCard movie={{ imdbID: 'tt00000', Title: 'The Matrix', Year: '1999', Poster: 'img', Type: 'Film' }} /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});
