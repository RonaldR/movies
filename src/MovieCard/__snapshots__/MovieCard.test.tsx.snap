// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie card 1`] = `
<DocumentFragment>
  <li
    class="movie-card"
  >
    <a
      class="movie-card__link"
      href="/detail/tt00000"
    >
      <figure
        class="movie-card__poster"
      >
        <img
          alt=""
          src="img"
        />
      </figure>
      <div
        class="movie-card__title"
      >
        The Matrix (1999)
      </div>
    </a>
  </li>
</DocumentFragment>
`;
